### DISCUSSION TITLE
Want to add a client blacklist mechanism

### DISCUSSION THREAD

**Discussion by XianzhiChris** [Tribler/tribler#6659] [Ideas]
Clients in the blacklist are prohibited from uploading
China's special network environment
Without this feature, Chinese users will not use the software

Or add a scoring mechanism to automatically block users who only download but do not upload

**Comment by ichorid** [comment]
How would you block them? By IP number and port? Or by Torrent client ID?
How other torrent clients do this in China?

**Reply by XianzhiChris** [reply]
Commonly used BT software, BitComet
http://wiki.bitcomet.com/bitcomet_options

“List of BitTorrent clients allowed connecting to”: Click the button “Client list” and tick the checkbox to choose to connect to a client or not.

Enable Anti-Leech: “Auto” by default.
Enable Anti-Leech will automatically refuse leeches' connection so as to improve the download rate. BT task Anti-Leech setting is divided to three levels:
Disable: Disable anti-leech.
Auto: Auto-start anti-leech when the parameters in “Advanced” settings, fail to be met.
Strict: Forced disconnect for some leeches and “Auto” for the others.

**Reply by XianzhiChris** [reply]
When using BT in China, you need to block the Xunlei client, because this software only downloads and will not upload and share

**Reply by XianzhiChris** [reply]
Long-Term Seeding It's also a great feature

Many torrent downloads are stuck at 99% because the seeder has left. BitComet tries to improve the performance by using the unique Long-Term Seeding Technology. This technology can find more seeds to help finish your downloads.

**Reply by ichorid** [reply]
We're using Libtorrent, so other clients based on it (e.g. qBittorrent) seem to have this problem too. It is much harder than it looks like:
https://github.com/qbittorrent/qBittorrent/issues/10258

**Reply by XianzhiChris** [reply]
The simple way is to let users shield themselves through the client name

**Reply by ichorid** [reply]
They change the name constantly

**Reply by XianzhiChris** [reply]
It usually contains XL logo
You can give it to the user to fill in
